src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompilationInfo.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.RegularExpressions
+{
+    public class RegexCompilationInfo
+    {
+        private string _pattern;
+        private string _name;
+        private string _nspace;
+        private TimeSpan _matchTimeout;
+
+        public RegexCompilationInfo(string pattern, RegexOptions options, string name, string fullnamespace, bool ispublic)
+            : this(pattern, options, name, fullnamespace, ispublic, Regex.DefaultMatchTimeout)
+        {
+        }
+
+        public RegexCompilationInfo(string pattern, RegexOptions options, string name, string fullnamespace, bool ispublic, TimeSpan matchTimeout)
+        {
+            Pattern = pattern;
```

Comment:
This is going to throw an ArgumentNullException with the name "value" rather than "pattern".  Is that what netfx does?  Does it match whatever input is provided via the public API that ends up here?

Reply:
Now that I'm comparing against the correct source, I see that this is what netfx does.  Ok.

Reply:
Looks like we had the same comment. It is what netfx does, but I think we should fix it.